### ISSUE TITLE
Cocoa Mac development support

### ISSUE THREAD

**Issue by mattheworiordan** [ably/ably-ios#460]
When trying to use this library for Mac development, the following error is shown:

```
"The platform of the target `iris` (OS X 10.11) is not compatible with `AblyRealtime (0.8.2)`, which does not support `osx`."
```

We need to ensure the library is compatible with OSX

**Comment by hemezh** [comment]
Thanks @mattheworiordan for the quick response.

**Comment by ricardopereira** [comment]
**For macOS compability, please have in mind:**
 - Reachability
 - Locale changes
 - Date and Time changes

**Comment by mattheworiordan** [comment]
@ricardopereira not something to look at now, but any idea how much work it may be to add Mac OS support?

**Comment by ricardopereira** [comment]
Adding a new framework target for the other platform and fix incompatibilities should be enough. I would say 2 days of work.

**Comment by mattheworiordan** [comment]
Ok, let's review once 0.9 is done

**Comment by rynosoft** [comment]
Did anything come of this?

**Comment by funkyboy** [comment]
@rynosoft Not yet. It shouldn't be a complicated task. Here's the plan I'd follow:

- import `AppKit` instead of `UIKit` if the target platform is macOS
- configure the test suite to run also on the `macOS` platform (Sierra and High Sierra).
- build a sample macOS app to make sure that our dependencies (KSCrash, msgpack and SocketRocket) work correctly also on macOS with both Carthage and Cocoapods.

@rynosoft Feel free to take a stab at it if you like :) 
Next week I plan to work on this for a few hours to see how far I get.

**Comment by rynosoft** [comment]
Hi Cesare:

I am taking a stab at it today. The first problem I’ve run into is that the Carthage update fails on msgpack. I’m researching for a solution. I’ll keep you apprised of my progress.

/Mick

**Comment by rynosoft** [comment]
The carthage update fails for msgpack. I tweeted at the author about Mac support.

**Comment by rynosoft** [comment]
I have successfully added Mac targets for the framework and tests. The tests appear to run successfully. I'm going to build a sample app next per your suggestion. Once I get it all working, I'll fork and push it.

**Comment by funkyboy** [comment]
@rynosoft Cool. So in the end `msgpack` works with macOS as a target?

**Comment by rynosoft** [comment]
I had to do a little work with msgpack which I'll also need to commit. Right now it's all cobbled together and won't work with carthage or cocoapods. I'll try to get that worked out in the next couple of days.

**Comment by funkyboy** [comment]
@rynosoft 👍 FYI, there are a bunch of [hurdles in pushing a new version of msgpack on Cocoapods](https://github.com/rvi/msgpack-objective-C/pull/3)

**Comment by rynosoft** [comment]
AFAIK, Cocoapods isn't being used for msgpack here. The three external frameworks (msgpack, KSCrash, SocketRocket) used to build Ably.framework are all pulled in by Carthage. The frameworks that are pulled by Cocoapods are just for the testing target.

**Comment by funkyboy** [comment]
@rynosof that is correct in the “I am developing the Ably SDK” context. 

But in the “I am using the Ably SDK to build an app and I am using Cocoapods as a dependency manager” users will just specify

`pod “Ably”`

and that will check the .podspec file, which in turn will instruct Cocopods to pull the `msgpack` pod.

**Comment by rynosoft** [comment]
Right! Sorry, I've only been the former until very recently. :)

**Comment by funkyboy** [comment]
No worries. I had the same thought when I got started :)

**Comment by benstahl** [comment]
I'm also very interested in using Ably on macOS, thanks for working on it and please let us know when there's code to use (hopefully via Carthage).

**Comment by rynosoft** [comment]
Hey all, I've finally got some bandwidth to work on this. I've submitted a pull request for msgpack. I'll follow with a pull request here when that is approved.

**Comment by mattheworiordan** [comment]
@rynosoft that's awesome. Is there anything we can do to help you at all?

**Comment by rynosoft** [comment]
I'll let you know if there is!

**Comment by rynosoft** [comment]
I just noticed that my pull request did not pass some tests using Travis. Before I delve into it, do you have any thoughts on what I can do?

https://travis-ci.org/rvi/msgpack-objective-C/builds/402354135?utm_source=github_status&utm_medium=notification

**Comment by funkyboy** [comment]
@rynosoft looks like an issue with the project configuration. Might want to get in touch with https://github.com/rvi about it, he's maintaining that project.

**Comment by ricardopereira** [comment]
[macOS is now supported](https://github.com/ably/ably-ios/blob/cfd33ea564f827b7847ed8bb0e309fb8657ef732/README.md#%EF%B8%8F-macos) but still in review and to be released.

**Comment by ricardopereira** [comment]
Released [v1.1.3](https://github.com/ably/ably-ios/releases/tag/1.1.3).
